Orange/widgets/data/owmergedata.py

```diff
@@ -83,32 +86,48 @@ def _setAttrs(self, model, data, othermodel, otherdata):
                 if len(model_) and model_[0] != INSTANCEID:
                     model_.insert(0, INSTANCEID)
 
+    def _find_best_match(self):
+        if self.dataA and self.dataB:
+            attr_a, attr_b, n_max_intersect = INDEX, INDEX, 0
+            str_metas_a = [m for m in self.dataA.domain.metas if m.is_string]
+            str_metas_b = [m for m in self.dataB.domain.metas if m.is_string]
+            for m_a, m_b in itertools.product(str_metas_a, str_metas_b):
+                n_inter = len(numpy.intersect1d(self.dataA[:, m_a].metas,
+                                                self.dataB[:, m_b].metas))
```

Comment:
What about another heuristic: the matching attributes will tend to have the same name. This may be a stupid idea, but perhaps you can add `+ (m_a.name == m_b.name)` just to break ties.